lib/catalog/rbac/group.rb

```diff
@@ -0,0 +1,23 @@
+module Catalog
```

Comment:
@eclarizio Can't we go back to the way it was done originally where the sharing/unsharing service would just call validate_groups. This seems to add another class, which basically does the same thing

Reply:
I would rather have the code that is calling out to RBAC contained in a single-purpose class so that if something about that needs to change, we only do it in one place. It's also easier to test since we don't have to mock out RBAC API calls in the share/unshare resource classes, we just mock out this class instead.

Yes, it does the same thing, but the concerns are separated, and I think that decoupling makes it cleaner overall.

Reply:
@eclarizio 
All the RBAC stuff is pretty isolated in the common gem.
https://github.com/RedHatInsights/insights-api-common-rails/blob/31b0dc49252c055cb41a8d4d63a82e84da91e876/lib/insights/api/common/rbac/utilities.rb#L6
If we have to make changes to RBAC this shared code has to change and this code is going to change when the RBAC team adds support for filtering by group ids.
It seems like the new class that is added is just wrapping another layer of complexity that we could do without.

we have the
(1) The RBAC API Client Gem
(2) The Insights API Common Gem
(3) This new class

To do a simple call I am not sure why it has to be nested 3 levels deep.

Reply:
> If we have to make changes to RBAC this shared code has to change and this code is going to change when the RBAC team adds support for filtering by group ids.

Seems like a good reason to have it separate, to me. Then we only have to change it in one place instead of all over in our code everywhere we'd be calling out to RBAC. I know I said it before but It also makes testing easier because it's only one place you actually need to test hitting the RBAC endpoint, and all of the rest of the specs can stub out our class that we know works. In my mind, approaching it from the point of testing and saying "wow this is easy to test" means the code is closer to having a single responsibility, which I think again makes for cleaner and more maintainable code in the long run.

Yes, it's another layer, but I would rather have a bunch of decoupled services that are flexible than tightly coupled classes that shouldn't share the same knowledge of how to do something. Basically, I'm saying I don't trust anything we don't own (including the Insights API Common Gem).